legend-depot-store-notifications/src/main/java/org/finos/legend/depot/store/notifications/store/mongo/NotificationsQueueMongo.java

```diff
@@ -113,15 +116,23 @@ public List<MetadataNotification> pullAll()
     }
 
     @Override
-    public Optional<MetadataNotification> getFirstInQueue()
+    public Optional<MetadataNotification> getFirstInQueueByPriority()
     {
-        Document first = (Document)getCollection().findOneAndDelete(Filters.exists("_id"),new FindOneAndDeleteOptions().sort(Sorts.ascending("createdAt")));
+        Document first = (Document)getCollection().findOneAndDelete(Filters.and(Filters.exists("_id"), Filters.eq("eventPriority", EventPriority.HIGH.name())), new FindOneAndDeleteOptions().sort(Sorts.ascending("createdAt")));
         if (first != null)
         {
             return Optional.of(convert(first, MetadataNotification.class));
 
         }
-        return Optional.empty();
+        else
```

Comment:
u dont need this. if there is nothing the queue rerturn empty
the query above will handle the scenario